staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/controller_ref.go

```diff
@@ -0,0 +1,41 @@
+package v1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// IsOwnedBy checks if the  object has a controllerRef set to the given owner
+func IsOwnedBy(obj Object, owner Object) bool {
+	ref := GetControllerOf(obj)
+	if ref == nil {
+		return false
+	}
+	return ref.UID == owner.GetUID()
+}
+
+// GetControllerOf returns the controllerRef if controllee has a controller,
+// otherwise returns nil.
+func GetControllerOf(controllee Object) *OwnerReference {
+	ownerRefs := controllee.GetOwnerReferences()
+	for i := range ownerRefs {
+		owner := &ownerRefs[i]
+		if owner.Controller != nil && *owner.Controller == true {
+			return owner
+		}
+	}
+	return nil
+}
```

Comment:
Yes, both should work. Maybe at first i intended to return the original ref, but its not the intention now.

I think your version is better, it's always returning a pointer to a copy, no matter how `GetOwnerReferences` is implemented.

Maybe we can update the comment to "GetControllerOf returns a pointer to a copy of the controllerRef if controllee has a controller"?